app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt

```diff
@@ -101,12 +101,14 @@ class FeatureFlagNotificationViewModel @Inject constructor(
     }
 
     private suspend fun setGuestRoomLinkFeatureFlag() {
-        observeGuestRoomLinkFeatureFlag().collect { guestRoomLinkStatus ->
-            guestRoomLinkStatus.isGuestRoomLinkEnabled?.let {
-                featureFlagState = featureFlagState.copy(isGuestRoomLinkEnabled = it)
-            }
-            guestRoomLinkStatus.isStatusChanged?.let {
-                featureFlagState = featureFlagState.copy(shouldShowGuestRoomLinkDialog = it)
+        viewModelScope.launch {
+            observeGuestRoomLinkFeatureFlag().collect { guestRoomLinkStatus ->
+                guestRoomLinkStatus.isGuestRoomLinkEnabled?.let {
+                    featureFlagState = featureFlagState.copy(isGuestRoomLinkEnabled = it)
+                }
+                guestRoomLinkStatus.isStatusChanged?.let {
+                    featureFlagState = featureFlagState.copy(shouldShowGuestRoomLinkDialog = it)
+                }
```

Comment:
In any case, inside `loadInitialSync` we're still calling a `collect {}` inside a `collect {}`.

That is, every time `Live` is received, we're adding another collector to the `guestRoomLinkFeatureFlag`.

We should replace `observeSyncState().collect` with either:
- `observeSyncState().collectLatest` if we're only interested in the latest value
- `observeSyncState().flatMapLatest` or `flatMapMerge` if we're interested in converting one value each another flow
- `observeSyncState().first { it == SyncState.Live }` if we just want to wait until the first time it is `Live`.

Reply:
You're right, I missed that, I think `collectLatest` could still produce multiple collectors, `flatMapLatest` would cancel all previous flows so it would replace the collector with every `LIVE`, but the last proposition with using `.first` is the easiest one so I changed the code to use this solution. 😄 Thanks!

Reply:
Multiple collectors indeed, I guess if making `cancellable().collect()` in this "internal collect" could fix it tho.

`first` seems fine to me too in any case. :)